### ISSUE TITLE
`verdi config list` errors when no profile is configured

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida-core#5920]
- [x] [AiiDA Troubleshooting Documentation](https://aiida.readthedocs.io/projects/aiida-core/en/latest/intro/troubleshooting.html)
- [x] [AiiDA Users Forum](https://groups.google.com/forum/#!forum/aiidausers)

### Describe the bug

When no profile is configured, I get

```
$ verdi config list
Traceback (most recent call last):
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/aiida/common/extendeddicts.py", line 51, in __getattr__
    return self[attr]
KeyError: 'profile'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/path/to/aiida/aiida/bin/verdi", line 8, in <module>
    sys.exit(verdi())
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/aiida/cmdline/commands/cmd_config.py", line 43, in verdi_config_list
    profile: Profile = ctx.obj.profile
  File "/path/to/aiida/aiida/lib/python3.9/site-packages/aiida/common/extendeddicts.py", line 54, in __getattr__
    raise AttributeError(errmsg)
AttributeError: 'AttributeDict' object has no attribute 'profile
```

### Steps to reproduce

Run `verdi config list` when no profile is configured

### Expected behavior

The command should exit gracefully, and probably even show the available options

### Your environment

- Operating system [e.g. Linux]: Linux
- Python version [e.g. 3.7.1]: 3.9
- aiida-core version [e.g. 1.2.1]: `main`

Other relevant software versions, e.g. Postres & RabbitMQ

### Additional context

@sphuber This is one of a class of issues that have to do with cli behavior when no profile is configured.
This is currently not tested since the profile is always already configured.
We may want to add a few "profile-free" tests for commands that we expect to still work when no profile is around.

**Comment by sphuber** [comment]
> @sphuber This is one of a class of issues that have to do with cli behavior when no profile is configured.
> This is currently not tested since the profile is always already configured.
> We may want to add a few "profile-free" tests for commands that we expect to still work when no profile is around.

Fair point. I already fixed a similar issue you reported some time ago #5543 . The infrastructure to easily test these is there, we just have to come up with the list of commands to test like this. I will start a PR and we can add them there.

**Comment by sphuber** [comment]
Turns out adding a test is not so easy. The problem is that the `ctx.obj.profile` object is not set when no profile is available. This is done by the custom `VerdiContext` which is invoked when `verdi` is invoked. However, in the testing we use the `run_cli_command` fixture, which doesn't actually go through `verdi` but goes straight to the subcommand, thereby circumventing the initialization of `ctx.obj` through the `VerdiContext`. This is "fixed" by having `run_cli_command` build the `ctx.obj` manually and pass it to the `CliRunner.invoke`. The problem then is, however, that the test will _always_ pass, so it is useless as a regression test.

One solution would be to run the test through an actual subprocess. I started this approach in #5846  . But then we cannot use the `empty_config` fixture, as that only operates in memory 😖 

Long story short, I removed the good-first-issue label for now.

**Comment by ltalirz** [comment]
Thanks for the investigation - indeed, in that case not a good-first issue ;-)





> This is "fixed" by having `run_cli_command` build the `ctx.obj` manually and pass it to the `CliRunner.invoke`

Could we just tell `run_cli_command` not to set `ctx.obj.profile` for such tests?

**Comment by ltalirz** [comment]
Anyhow, it's not a priority; no need to lose a lot of time over this.
Once #5846 is ready for review, let me know

**Comment by unkcpz** [comment]
I just run into the same issue, is #5846 will fix this? Regardless of the test, adding `@decorators.with_dbenv()` for `verdi_config_list` should fix this. Is a separate PR welcome?

**Comment by ltalirz** [comment]
`verdi config` shouldn't require the dbenv though, should it?

**Comment by unkcpz** [comment]
> verdi config shouldn't require the dbenv though, should it?

I am not sure, I think it depends on if there are any config settings store in the DB, if not it shouldn't require.
